# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CodeGenerator.cpp
# Commit: 349728105c40
# Full Hash: 349728105c40993896170f4c2f43e152a3466706
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-01-08 09:40:24
# Description:
#   Bug 1748602 - Backout bug 1642412. r=lth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D135346
# ==============================================================================

diff -r 32873022d2ed -r 349728105c40 js/src/jit/CodeGenerator.cpp
--- a/js/src/jit/CodeGenerator.cpp	Fri Jan 07 20:35:15 2022 +0000
+++ b/js/src/jit/CodeGenerator.cpp	Fri Jan 07 20:43:16 2022 +0000
@@ -8178,26 +8178,11 @@
   }
 }
 
-void CodeGenerator::visitWasmLoadTableElement(LWasmLoadTableElement* ins) {
-  Register elements = ToRegister(ins->elements());
-  Register index = ToRegister(ins->index());
-  Register output = ToRegister(ins->output());
-  masm.loadPtr(BaseIndex(elements, index, ScalePointer), output);
-}
-
 void CodeGenerator::visitWasmDerivedPointer(LWasmDerivedPointer* ins) {
   masm.movePtr(ToRegister(ins->base()), ToRegister(ins->output()));
   masm.addPtr(Imm32(int32_t(ins->offset())), ToRegister(ins->output()));
 }
 
-void CodeGenerator::visitWasmDerivedIndexPointer(
-    LWasmDerivedIndexPointer* ins) {
-  Register base = ToRegister(ins->base());
-  Register index = ToRegister(ins->index());
-  Register output = ToRegister(ins->output());
-  masm.computeEffectiveAddress(BaseIndex(base, index, ins->scale()), output);
-}
-
 void CodeGenerator::visitWasmStoreRef(LWasmStoreRef* ins) {
   Register tls = ToRegister(ins->tls());
   Register valueAddr = ToRegister(ins->valueAddr());